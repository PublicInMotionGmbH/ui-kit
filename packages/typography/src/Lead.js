import React from 'react'
import PropTypes from 'prop-types'
import cls from 'classnames'

import { prefix } from '@talixo/shared'

const moduleName = prefix('lead')

/**
 * Component which represents article/page lead.
 *
 * @param {object} props
 * @param {string} [props.className]
 * @param {*} [props.children]
 * @returns {React.Element}
 */
function Lead (props) {
  const { className, children, ...passedProps } = props

  return (
    <p className={cls(moduleName, className)} {...passedProps}>
      {children}
    </p>
  )
}

Lead.propTypes = {
  /** Additional class name */
  className: PropTypes.string,

  /** Lead content */
  children: PropTypes.node
}

export default Lead
