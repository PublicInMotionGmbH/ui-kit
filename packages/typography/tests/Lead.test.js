import React from 'react'
import { shallow } from 'enzyme'

import Lead from '../src/Lead'

describe('<Lead />', () => {
  it('renders empty correctly', () => {
    const wrapper = shallow(<Lead />)

    expect(wrapper).toMatchSnapshot()
  })

  it('renders children correctly', () => {
    const wrapper = shallow(
      <Lead>We are very proud of how does it work.</Lead>
    )

    expect(wrapper).toMatchSnapshot()
  })
})
