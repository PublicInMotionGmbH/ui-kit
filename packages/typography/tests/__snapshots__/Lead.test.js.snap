// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Lead /> renders children correctly 1`] = `
<p
  className="talixo-lead"
>
  We are very proud of how does it work.
</p>
`;

exports[`<Lead /> renders empty correctly 1`] = `
<p
  className="talixo-lead"
/>
`;
